zerver/lib/integrations.py

```diff
@@ -186,6 +186,40 @@ def url_object(self):
         # type: () -> None
         return
 
+class EmbeddedBotService(Integration):
+    '''
+    This class acts as a registry for bots verified as safe
+    and valid such that these are capable of being deployed on the server.
+    '''
+    IS_ALLOWED = True
+    # This variable informs if this embedded bot service is allowed to run by this
+    # community members. The Organization administrators can turn this off if they
+    # do not want community members to run this bot service.
+
+    def __init__(self, name):
+        # type: (str) -> None
+        super(EmbeddedBotService, self).__init__(
+            name=name,
+            client_name=name,
+            # TODO: Not sure how client_name would be useful here, but it is not None parameter.
+            # setting it to name of the bot for now.
+            categories=['deployment']
+        )
+
+    def is_enabled(self):
+        # type: () -> bool
+        # Not ready to be added to the docs yet.
+        return False
+
+    def is_allowed(self, allow):
+        # type: (bool) -> None
+        self.IS_ALLOWED = allow
```

Comment:
I'm not sure why we need `IS_ALLOWED`; I was imagining we'd just add them to `EMBEDDED_BOT_SERVICES` or not based on whether they're allowed...